rfc/2020/conformance-validation/mayankshah1607/rfc.md

```diff
@@ -0,0 +1,443 @@
+- Contribution Name: Conformance Validation
+- Implementation Owner: [@mayankshah1607](https://github.com/mayankshah1607)
+- Start Date: 2020-02-23 (24th Feb, 2020)
+- Target Date: 
+- RFC PR:
+- Linkerd Issue: https://github.com/linkerd/linkerd2/issues/1096
+- Reviewers: 
+
+# Summary
+
+[summary]: #summary
+This project proposes a new test suite that shall be used for conformance validation. The new test suite shall be used to validate non-trivial network communication (HTTP, gRPC, websocket) among stateless and stateful workloads in the Linkerd data plane. The correctness of the interaction between the Linkerd control plane and the Kubernetes API Server will also be tested. This shall be done by carrying out extensive e2e tests of Linkerd features using a sample distributed application (data plane).
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+Linkerd has an extensive check suite that validates a cluster is ready to install Linkerd and that the installation was successful. These checks are, unfortunately, static checks. Because of the wide number of ways a Kubernetes cluster can be configured, users want a way to validate their specific install for conformance over time. The proposed project tackles this problem by allowing users to deploy sample workloads to their cluster and carry out extensive e2e (conformance) tests. 
+
+# Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+## Interaction with other features
+The e2e conformance tests proposed would be carried out outside the Linkerd CI. These tests would make use of the `kubectl` and `linkerd` binaries to interact with the cluster and with various features Linkerd can perform.
+
+We initially start off by modifying the existing emojivoto application to have feature flags to enable/disable features such as MySQL, Redis, gRPC, websockets, etc. This is important because emojivoto is heavily used in the getting started process.
```

Comment:
A Note to remember: Care has to be taken to make these features that we will add are independent of each other, and would not fail the application, when any of the flag is turned off i.e Ideally each new protocol/feature should be independent of the normal working of the application